extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3Utils.java

```diff
@@ -214,14 +217,14 @@ static void uploadFileIfPossible(
       String bucket,
       String key,
       File file
-  )
+  ) throws java.lang.InterruptedException
```

Comment:
where does this exception get caught now? it's usually not a good idea to catch interrupted exceptions and ignoring / just logging them, so we need to make sure this is handled properly

Reply:
> where does this exception get caught now? it's usually not a good idea to catch interrupted exceptions and ignoring / just logging them, so we need to make sure this is handled properly

All the S3Utils.uploadFileIfPossible() calls are wrapped within S3Utils.retryS3Operation() such as the following in S3DataSegmentPusher.java:
```
    try {
      return S3Utils.retryS3Operation(
          () -> {
            S3Utils.uploadFileIfPossible(s3Client, config.getDisableAcl(), config.getBucket(), s3Path, zipOutFile);

            return outSegment;
          }
      );
    }
    catch (AmazonServiceException e) {
      throw new IOException(e);
    }
    catch (Exception e) {
      throw new RuntimeException(e);
    }
    finally {
      log.debug("Deleting temporary cached index.zip");
      zipOutFile.delete();
    }
```

Reply:
we'll want to catch the interrupted exception explicitly and reset the interrupt flag to make sure it is not lost. Here's an article that explains what I mean and what needs how to handle this https://dzone.com/articles/how-to-handle-the-interruptedexception

Reply:
> we'll want to catch the interrupted exception explicitly and reset the interrupt flag to make sure it is not lost. Here's an article that explains what I mean and how to handle this https://dzone.com/articles/how-to-handle-the-interruptedexception

Thanks, I have updated the code. I read the article https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html and use Thread.interrupted() to clear the state.